### PR TITLE
Tag system

### PR DESCRIPTION
**Pull request by adrianob**
Simple comma separated tag system for admin use.

### CODE REVIEW

File: db/migrate/20151203184721_create_tags.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class CreateTags < ActiveRecord::Migration
+  def change
+    create_table :tags do |t|
+      t.text :name
+
+      t.timestamps
```
Comment (devton) [inline db/migrate/20151203184721_create_tags.rb]: @adrianob unique index on tag name?  ```add_index :tags, :name, unique: true```

File: db/migrate/20151203190446_create_taggings.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateTaggings < ActiveRecord::Migration
+  def change
+    create_table :taggings do |t|
+      t.references :tag
+      t.references :project
+
+      t.timestamps
```
Comment (devton) [inline db/migrate/20151203190446_create_taggings.rb]: scoped unique index here ```add_index :taggings, [:tag_id, :project_id], unique: true```

File: app/models/project.rb [hunk 3]
```diff
@@ -293,6 +295,16 @@ def open_for_contributions?
     pluck_from_database(:open_for_contributions)
   end
 
+  def all_tags=(names)
+    self.tags = names.split(',').map do |name|
+      Tag.find_or_create_by(name: name.strip)
```
Comment (devton) [inline app/models/project.rb]: lets force lower case on tag name?

File: app/models/project.rb [hunk 4]
```diff
@@ -293,6 +295,18 @@ def open_for_contributions?
     pluck_from_database(:open_for_contributions)
   end
 
+  def all_tags=(names)
+    self.tags = names.split(',').map do |name|
+      Tag.find_or_create_by(name: name.strip) do |tag|
```
Comment (devton) [inline app/models/project.rb]: @adrianob we should use ```find_or_create_by(slug: ...)``` here not?

Comment (adrianob) [inline app/models/project.rb]: The idea is to save the exact user input and a slug to help searching later, if we look for slug first there is no way to recover the information later, that is, there is no way of going from slug to name.

**Pull request merged**